scripts/obtain/install.ps1

```diff
@@ -3,106 +3,219 @@
-    if ($LocalVersion -and $LocalHash)
-    {
-        if ($Version -eq "Latest")
-        {
-            $RemoteResponse = Invoke-WebRequest -UseBasicParsing "$Feed/$Channel/dnvm/latest.win.$Architecture.version"
-            $RemoteData = @([Text.Encoding]::UTF8.GetString($RemoteResponse.Content).Split([char[]]@(), [StringSplitOptions]::RemoveEmptyEntries));
-            $RemoteHash = $RemoteData[0].Trim()
-            $RemoteVersion = $RemoteData[1].Trim()
-
-            if (!$RemoteVersion -or !$RemoteHash) {
-                throw "Invalid response from feed"
-            }
-
-            say "Latest version: $RemoteVersion"
-            say "Local Version: $LocalVersion"
-
-            if($LocalHash -eq $RemoteHash)
-            {
-                say "You already have the latest version"
-                exit 0
-            }
-        }
-        elseif ($LocalVersion -eq $Version)
-        {
-            say "$Version is already installed."
-            exit 0
+function Say-Verbose($str) {
+    Write-Verbose "dotnet_install: $str"
+}
+
+function Say-Invocation($Invocation) {
+    $command = $Invocation.MyCommand;
+    $args = (($Invocation.BoundParameters.Keys | foreach { "-$_ `"$($Invocation.BoundParameters[$_])`"" }) -join " ")
+    Say-Verbose "$command $args"
+}
+
+function Get-Machine-Architecture() {
+    Say-Invocation $MyInvocation
+
+    # possible values: AMD64, IA64, x86
+    return $ENV:PROCESSOR_ARCHITECTURE
+}
+
+# TODO: Architecture and CLIArchitecture should be unified
+function Get-CLIArchitecture-From-Architecture([string]$Architecture) {
+    Say-Invocation $MyInvocation
+
+    switch ($Architecture.ToLower()) {
+        { $_ -eq "auto" } { return Get-CLIArchitecture-From-Architecture $(Get-Machine-Architecture) }
+        { ($_ -eq "amd64") -Or ($_ -eq "x64") } { return "x64" }
+        { $_ -eq "x86" } { return "x86" }
+        default { throw "Architecture not supported. If you think this is a bug, please report it at https://github.com/dotnet/cli/issues" }
+    }
+}
+
+function Get-Version-Info-From-Version-Text([string]$VersionText) {
+    Say-Invocation $MyInvocation
+
+    $Data = @($VersionText.Split([char[]]@(), [StringSplitOptions]::RemoveEmptyEntries));
+
+    $VersionInfo = @{}
+    $VersionInfo.CommitHash = $Data[0].Trim()
+    $VersionInfo.Version = $Data[1].Trim()
+    return $VersionInfo
+}
+
+function Get-Latest-Version-Info([string]$AzureFeed, [string]$AzureChannel, [string]$CLIArchitecture) {
+    Say-Invocation $MyInvocation
+
+    $VersionFileUrl = "$AzureFeed/$AzureChannel/dnvm/latest.win.$CLIArchitecture.version"
+    $Response = Invoke-WebRequest -UseBasicParsing $VersionFileUrl
+    $VersionText = [Text.Encoding]::UTF8.GetString($Response.Content)
+
+    $VersionInfo = Get-Version-Info-From-Version-Text $VersionText
+
+    return $VersionInfo
+}
+
+# TODO: AzureChannel and Channel should be unified
+function Get-Azure-Channel-From-Channel([string]$Channel) {
+    Say-Invocation $MyInvocation
```

Comment:
`RELEASE_SUFFIX` is used as channel in our build scripts. https://github.com/dotnet/cli/search?utf8=%E2%9C%93&q=RELEASE_SUFFIX

You need to update that to reflect these new values `nightly`, `preview`.